src/Microsoft.Extensions.Caching.Abstractions/IMemoryCache.cs

```diff
@@ -30,5 +30,11 @@ public interface IMemoryCache : IDisposable
         /// </summary>
         /// <param name="key">An object identifying the entry.</param>
         void Remove(object key);
+
+        /// <summary>
+        /// Attempt to remove the given percentage of the total entries in the cache.
+        /// </summary>
+        /// <param name="percentage">The percentage of cache entries to remove.</param>
+        void Compact(double percentage);
```

Comment:
Also, since we are exposing Compact, should we expose Count on the interface as well? It's not a good way to gauge cache size but I suspect users will downcast and rely on it to determine how much to compact like in https://github.com/aspnet/Caching/pull/252.